src/syscall/syscall_libc.go

```diff
@@ -290,6 +292,18 @@ func Environ() []string {
 	return envs
 }
 
+// BytePtrFromString returns a pointer to a NUL-terminated array of
+// bytes containing the text of s. If s contains a NUL byte at any
+// location, it returns (nil, EINVAL).
+func BytePtrFromString(s string) (*byte, error) {
```

Comment:
Might need to special-case `len(s) == 0` to avoid panic on `[0]` like in #3303

Reply:
`cstring(s)` will append a null byte, so it will always be at least one byte in length. Such a panic shouldn't be possible.